mod/cornercutter/SpawnCollection.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace cornercutter
+{
+    class SpawnCollection
+    {
+        public List<WeightedSkill> Skills { get; private set; }
+        // Boolean for whether we should wrap around when pulling this pool or treat it as a bucket to grab from
+        private readonly bool isLooping;
+        // For now, where isLooping is set, currentIndex is used, otherwise currentTotal for weight tracking
+        // Seperated for clarity
+        private int currentIndex;
+        private int currentTotal;
+        private readonly Random rng;
+
+        public SpawnCollection(bool loopSpawns)
+        {
+            Skills = new List<WeightedSkill>();
+            currentIndex = 0;
+            isLooping = loopSpawns;
+            if (isLooping)
+            {
+                // Only instantiate if needed
+                rng = new Random();
+            }
+        }
+
+        public void AddSkill(WeightedSkill skill)
+        {
+            Skills.Add(skill);
+            if (isLooping)
+            {
+                currentTotal += skill.Weight;
+            }
+        }
+
+        public void AddSkills(params WeightedSkill[] skills)
+        {
+            foreach (WeightedSkill skill in skills)
+            {
+                AddSkill(skill);
+            }
+        }
+
+        public Entity GetNextSkill()
+        {
+            if (Skills.Count == 0) return null;
+
+            Entity selectedSkill = null;

```

Comment:
It seems like you've accounted for the `Looped` and `Weighted` spawn types, but what about `Consecutive`? I assume this would be similar to looped except once it runs out of skills it just returns nothing? 

Maybe using a Factory design pattern with each of these different spawn types as different subclasses of some `ISpawnCollection` interface may make it easier to manage?

Reply:
Yeah, good point - this class was not originally written with `Consecutive ` in mind, but even with that addition, it becomes clear this should be written a bit more generically - for example, if a spawner that removed items from the pool was added, how would that be weaved in? As such, I will rework this in the next commit.